### PR TITLE
Bump phpstan

### PR DESCRIPTION
**Pull request by VincentLanglet**
The last error must be a phpstan issue
https://github.com/phpstan/phpstan/issues/5893

## Changelog 

```markdown
### Deprecated
- `FormTypeFieldExtension::getValueFromFieldDescription()`
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
Maybe we can ignore that last error until it gets fixed by phpstan or proved wrong?

**Comment by VincentLanglet** [comment]
> Maybe we can ignore that last error until it gets fixed by phpstan or proved wrong?

Done, Ready for review @sonata-project/contributors

File: src/Form/Extension/Field/Type/FormTypeFieldExtension.php [hunk 1]
```diff
@@ -182,13 +185,22 @@ public function configureOptions(OptionsResolver $resolver): void
...
+     * @deprecated since sonata-project/admin-bundle version 4.x and will be removed in 5.0.
+     *
      * return the value related to FieldDescription, if the associated object does no
      * exists => a temporary one is created.
      *
      * @return mixed
      */
     public function getValueFromFieldDescription(?object $object, FieldDescriptionInterface $fieldDescription)
     {
+        @trigger_error(sprintf(
```
Comment (jordisala1991) [inline src/Form/Extension/Field/Type/FormTypeFieldExtension.php:199]: shouldn't this get deprecated in another PR? otherwise this is not pedantic but minor instead. and deserves a changelog aswell as upgrade note

Comment (VincentLanglet) [reply src/Form/Extension/Field/Type/FormTypeFieldExtension.php:199]: I added the changelog

Comment (jordisala1991) [reply src/Form/Extension/Field/Type/FormTypeFieldExtension.php:199]: Can you explain why you want to deprecate this, and what are the replacements if any? Also, can you add an entry on the upgrade 4.x file?

**Review by phansys — APPROVED** [review]

**Review by core23 — APPROVED** [review]

File: src/Form/Extension/Field/Type/FormTypeFieldExtension.php [hunk 2]
```diff
@@ -54,8 +54,8 @@ public function __construct(array $defaultClasses, array $options)
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         $sonataAdmin = [
-            'name' => null,
-            'admin' => null,
+            'name' => false,
```
Comment (core23) [inline src/Form/Extension/Field/Type/FormTypeFieldExtension.php:57]: You should revert this change @VincentLanglet 

This line is breaking with a `false` value:
https://github.com/sonata-project/SonataAdminBundle/blob/0639d128cb0516584ab435c0a726dd205ded8ef9/src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig#L100-L101

Comment (VincentLanglet) [reply src/Form/Extension/Field/Type/FormTypeFieldExtension.php:57]: Shouldn't we change the check to 
```
{% if sonata_admin.admin %}
```
instead ?

Either false or null shouldn't go in the if.
And we were mixing both null and false value before. I tried to always use `false` instead.
```
{% if sonata_admin.admin is not false %}
```
may work now but 
```
{% if sonata_admin.admin %}
```
might be safer and more logic

Comment (VincentLanglet) [reply src/Form/Extension/Field/Type/FormTypeFieldExtension.php:57]: https://github.com/sonata-project/SonataAdminBundle/pull/7609
